inspire_dojson/hep/model.py

```diff
@@ -83,12 +83,37 @@ def convert_curated(record, blob):
     return record
 
 
+def ensure_ordered_figures(record, blob):
+    ordered_figures_dict = {}
+    unordered_figures_list = []
+
+    for figure in record.get('figures', []):
+        order = figure.get('order')
+        if order:
+            figure.pop('order')
+            ordered_figures_dict[order] = figure
+        else:
+            unordered_figures_list.append(figure)
+
+    if ordered_figures_dict:
+        sorted_keys = ordered_figures_dict.keys()
+        sorted_keys.sort()
+
+        ordered_figures_list = [ordered_figures_dict[key] for key in sorted_keys]
+        ordered_figures_list.append(unordered_figures_list)
+        record['figures'] = ordered_figures_list
+    else:
+        record['figures'] = unordered_figures_list
```

Comment:
and here you just do `record['figures'].extend(unordered_figures_list)`. So you skip the test, and in the case you have both ordered and unordered figures (that probably doesn't happen, but better safe than sorry), you don't throw away part of them.